plugin/src/test/java/org/opensearch/ml/rest/MLCommonsTenantAwareRestTestCase.java

```diff
@@ -0,0 +1,243 @@
+
+    protected final RestRequest tenantRequest = new FakeRestRequest.Builder(NamedXContentRegistry.EMPTY)
+        .withHeaders(Map.of(TENANT_ID_HEADER, singletonList(tenantId)))
+        .build();
+    protected final RestRequest otherTenantRequest = new FakeRestRequest.Builder(NamedXContentRegistry.EMPTY)
+        .withHeaders(Map.of(TENANT_ID_HEADER, singletonList(otherTenantId)))
+        .build();
+    protected final RestRequest nullTenantRequest = new FakeRestRequest.Builder(NamedXContentRegistry.EMPTY)
+        .withHeaders(emptyMap())
+        .build();
+
+    protected final RestRequest tenantMatchAllRequest = getRestRequestWithHeadersAndContent(tenantId, MATCH_ALL_QUERY);
+    protected final RestRequest otherTenantMatchAllRequest = getRestRequestWithHeadersAndContent(otherTenantId, MATCH_ALL_QUERY);
+    protected final RestRequest nullTenantMatchAllRequest = getRestRequestWithHeadersAndContent(null, MATCH_ALL_QUERY);
+
+    protected static boolean isMultiTenancyEnabled() throws IOException {
+        // pass -Dtests.rest.tenantaware=true on gradle command line to enable
+        return Boolean.parseBoolean(System.getProperty(ML_COMMONS_MULTI_TENANCY_ENABLED.getKey()))
+            || Boolean.parseBoolean(System.getenv(ML_COMMONS_MULTI_TENANCY_ENABLED.getKey()));
+    }
+
+    protected static Response makeRequest(RestRequest request, String method, String path) throws IOException {
+        return TestHelper
+            .makeRequest(client(), method, path, request.params(), request.content().utf8ToString(), getHeadersFromRequest(request));
+    }
+
+    private static List<Header> getHeadersFromRequest(RestRequest request) {
+        return request
+            .getHeaders()
+            .entrySet()
+            .stream()
+            .map(e -> new BasicHeader(e.getKey(), e.getValue().stream().collect(Collectors.joining(","))))
+            .collect(Collectors.toList());
+    }
+
+    protected static RestRequest getRestRequestWithHeadersAndContent(String tenantId, String requestContent) {
+        Map<String, List<String>> headers = new HashMap<>();
+        if (tenantId != null) {
+            headers.put(Constants.TENANT_ID_HEADER, singletonList(tenantId));
+        }
+        return new FakeRestRequest.Builder(NamedXContentRegistry.EMPTY)
+            .withHeaders(headers)
+            .withContent(new BytesArray(requestContent), JSON)
+            .build();
+    }
+
+    @SuppressWarnings("unchecked")
+    protected static Map<String, Object> responseToMap(Response response) throws IOException {
+        return parseResponseToMap(response);
+    }
+
+    @SuppressWarnings("unchecked")
+    protected static String getErrorReasonFromResponseMap(Map<String, Object> map) {
+        // Two possible cases:
+        String type = ((Map<String, String>) map.get("error")).get("type");
+
+        // {
+        // "error": {
+        // "root_cause": [
+        // {
+        // "type": "status_exception",
```

Comment:
Why it's showing 2 times?

Reply:
Do you know from where are we getting this `status_exception`? May be we can change that to `OpenSearchStatusException` too? I can make the changes.

Reply:
I think this is a difference between local vs. remote client exceptions.